### PR TITLE
Fix crash on terminal close

### LINKED ISSUE #4490: After last update lens Crashes repeatedly.

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: src/main/shell-session/shell-session.ts [hunk 1]
```diff
@@ -269,6 +269,10 @@ export abstract class ShellSession {
               break;
           }
         } catch (error) {
+          if (error?.message === "ioctl(2) failed, EBADF") {
```
Comment (jim-docker) [inline src/main/shell-session/shell-session.ts:272]: very specific, why bother?

**Review by jim-docker — COMMENTED** [review]
testing...

File: src/main/shell-session/shell-session.ts (hunk 1 above)
Comment (Nokel81) [reply src/main/shell-session/shell-session.ts:272]: Fair enough, I looked it up and it is a harmless error. But I'll revert.

**Comment by jakolehm** [comment]
Tested this on Windows 11, app still hangs after I close terminal tab.

**Comment by jakolehm** [comment]
App hangs only if I close terminal via "x":
```
[0] info:    ▪ [SHELL-SESSION]: Killing shell process (pid=64484) for 5e528ff3-1e9e-40ee-a135-8aae3a16c76e:4fb0af88-6340-427e-8298-d9c6640b5961 +0ms
[0] C:\Users\jarik\Code\lens\node_modules\node-pty\lib\conpty_console_list_agent.js:17
[0] var consoleProcessList = getConsoleProcessList(shellPid);
[0]                          ^
[0]
[0] Error: AttachConsole failed
[0]     at Object.<anonymous> (C:\Users\jarik\Code\lens\node_modules\node-pty\lib\conpty_console_list_agent.js:17:26)
[0]     at Module._compile (internal/modules/cjs/loader.js:1078:30)
[0]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1108:10)
[0]     at Module.load (internal/modules/cjs/loader.js:935:32)
[0]     at Module._load (internal/modules/cjs/loader.js:776:14)
[0]     at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
[0]     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
[0]     at internal/main/run_main_module.js:17:47
```

If I type `exit` then it works ok.

**Comment by Nokel81** [comment]
This does fix an unhandled exception but not the windows freeze issue

**Review by jim-docker — APPROVED** [review]
You should remove "fixes #4490" from the description? Is there an issue for what it does fix?

**Comment by Nokel81** [comment]
> You should remove "fixes #4490" from the description

Done

> Is there an issue for what it does fix?

No, it was asked about on slack and I have seen it, but I don't think an issue was made.

**Pull request closed without merging**